src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -726,6 +726,7 @@ private async Task<DataPackageOperation> StartDragAsyncCore(PointerPoint pointer
 			var routedArgs = new DragStartingEventArgs(this, ptArgs);
 			PrepareShare(routedArgs.Data); // Gives opportunity to the control to fulfill the data
 			SafeRaiseEvent(DragStartingEvent, routedArgs); // The event won't bubble, cf. PrepareManagedDragAndDropEventBubbling
+			UpdateLayout(); // we need a layout cycle before a possible DragEnter just in case DragStarting causes the visual tree to change
```

Comment:
this can be removed in favor of https://github.com/unoplatform/uno/pull/17357, which is a cleaner fix.